packages/kbn-apm-utils/src/index.ts

```diff
@@ -71,13 +73,17 @@ export async function withSpan<T>(
       span.addLabels(labels);
     }
 
-    return cb()
+    return cb(span)
       .then((res) => {
-        span.outcome = 'success';
+        if (!span.outcome || span.outcome === 'unknown') {
+          span.outcome = 'success';
```

Comment:
I know this is after the fact, but there is a `span.setOutcome(outcome)` [documented API](https://www.elastic.co/guide/en/apm/agent/nodejs/current/span-api.html#span-setoutcome) on APM span objects that will deal with avoiding "has this outcome already been set for other reasons" conflicts.

Reply:
Doesn't that always override? In this case I only want to set it when there is no explicit value yet.

Reply:
Okay. Yes, `span.setOutcome(outcome)` always overrides. I wasn't clear on the intent.